src/olympia/api/middleware.py

```diff
@@ -2,16 +2,22 @@
 
 from django.conf import settings
 from django.middleware.gzip import GZipMiddleware
+from django.utils.cache import patch_vary_headers
 from django.utils.deprecation import MiddlewareMixin
 
 
-class IdentifyAPIRequestMiddleware(MiddlewareMixin):
+class APIRequestMiddleware(MiddlewareMixin):
     def identify_request(self, request):
         request.is_api = re.match(settings.DRF_API_REGEX, request.path_info)
 
     def process_request(self, request):
         self.identify_request(request)
 
+    def process_response(self, request, response):
+        if request.is_api:
+            patch_vary_headers(response, ['X-Country-Code'])
```

Comment:
_all_ api requests?

Reply:
It's just a `Vary` on a header that shouldn't be sent very often, so I think it's safe

Reply:
It's a lot easier to add there - we might return the 451 for anything that depend on an add-on, but we don't have a centralized way of returning the `Vary` header otherwise